File: frontend/src/types/project-member.ts
export interface ProjectMember {
  id: string;
  project_id: string;
  user_id: string;
  user_name?: string;
  rate_function_id?: string;
  rate_function_name?: string;
  hourly_rate?: number;
  resolved_rate?: number;
  role_label?: string;
  budget_hours: number | null;
  joined_at: string;
  left_at?: string;
  created_at: string;
  updated_at: string;
}

export interface AddProjectMemberRequest {
  user_id: string;
  rate_function_id?: string;
  hourly_rate?: number;
  role_label?: string;
}

export interface UpdateProjectMemberRequest {
  rate_function_id?: string;
  hourly_rate?: number;
  role_label?: string;
  left_at?: string;
}

File: backend/crates/konto-api/src/dto/project_member.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(Debug, Clone, Serialize, ToSchema)]
pub struct ProjectMemberResponse {
    pub id: String,
    pub project_id: String,
    pub user_id: String,
    pub user_name: Option<String>,
    pub rate_function_id: Option<String>,
    pub rate_function_name: Option<String>,
    #[schema(value_type = Option<String>)]
    pub hourly_rate: Option<rust_decimal::Decimal>,
    #[schema(value_type = Option<String>)]
    pub resolved_rate: Option<rust_decimal::Decimal>,
    pub role_label: Option<String>,
    #[schema(value_type = Option<String>)]
    pub budget_hours: Option<rust_decimal::Decimal>,
    pub joined_at: String,
    pub left_at: Option<String>,
}

#[derive(Debug, Deserialize, ToSchema)]
pub struct CreateProjectMemberRequest {
    pub user_id: String,
    pub rate_function_id: Option<String>,
    #[schema(value_type = Option<String>)]
    pub hourly_rate: Option<rust_decimal::Decimal>,
    pub role_label: Option<String>,
    #[schema(value_type = Option<String>)]
    pub budget_hours: Option<rust_decimal::Decimal>,
}

#[derive(Debug, Deserialize, ToSchema)]
pub struct UpdateProjectMemberRequest {
    pub rate_function_id: Option<Option<String>>,
    #[schema(value_type = Option<Option<String>>)]
    pub hourly_rate: Option<Option<rust_decimal::Decimal>>,
    pub role_label: Option<Option<String>>,
    #[schema(value_type = Option<Option<String>>)]
    pub budget_hours: Option<Option<rust_decimal::Decimal>>,
}
